src/app_service/service/community/service.nim

```diff
@@ -563,6 +632,13 @@ QtObject:
   proc getCuratedCommunities*(self: Service): seq[CuratedCommunity] =
     return toSeq(self.curatedCommunities.values)
 
+  proc getCommunityByIdFromAllCommunities*(self: Service, communityId: string): CommunityDto =
```

Comment:
Surprised we didn't have that function before. Oh well lol.

Reply:
We were relying on `joinedCommunities` when we did things like `getCommunityByID`.
Unfortunately that list of communities won't do the trick when dealing with `spectatedCommunities` (which is what we have when looking at a community that we haven't joined yet and might be token gated).

This has actually bitten be twice very badly.

@osmaczko created and issue about this a while ago: https://github.com/status-im/status-desktop/issues/9570

Bottom line is that we should just have `allCommunities`

Reply:
The real bottom line imho is that (and this pertains to all our NIM models) we should never do any sorting/filtering/bucketing on the backend side. Just like here, if a community has a `joined` model role, we can easily filter by that in QML

Reply:
I believe the "bucketing" is a result of historic changes throughout the lifetime of the protocol.

But yes, we've already agreed in #9570 that we'll just have **one** list of known communities and then dedicated flags to know whether or not they are "joined" or "curated" or neither if necessary.